GitUI/GitUIExtensions.cs

```diff
@@ -86,11 +86,30 @@ public static Task ViewChanges(this FileViewer diffViewer, IList<GitRevision> re
 
         public static Task ViewChanges(this FileViewer diffViewer, string firstRevision, string secondRevision, GitItemStatus file, string defaultText)
         {
-            return diffViewer.ViewPatch(() =>
+            if (firstRevision == null)
             {
-                string selectedPatch = diffViewer.GetSelectedPatch(firstRevision, secondRevision, file);
-                return selectedPatch ?? defaultText;
-            });
+                // The previous commit does not exist, nothing to compare with
+                if (file.TreeGuid.IsNullOrEmpty())
+                {
+                    if (secondRevision.IsNullOrWhiteSpace())
+                    {
+                        throw new ArgumentException(nameof(secondRevision));
```

Comment:
For the release/2.51 merge, I propose that defaultText is returned here. Use of null for revisions need to be investigated in #4586

Reply:
Agree

Reply:
I prefer to fail to showing false data.